src/openloco/window.h

```diff
@@ -18,23 +31,37 @@ namespace openloco::ui
             uint8_t pad_all[0x88E];
             struct
             {
-                uint8_t pad_00[0x0C];
+                void* event_handlers;     // 0x00
+                void* viewport;           // 0x04
+                uint8_t pad_08[0x04];     // 0x08
                 uint32_t enabled_widgets; // 0x0C
                 uint8_t pad_10[0x2C - 0x10];
-                widget* widgets; // 0x2C
-                uint8_t pad_30[0x40 - 0x30];
+                widget* widgets;     // 0x2C
+                uint16_t x;          // 0x30
+                uint16_t y;          // 0x32
+                uint16_t width;      // 0x34
+                uint16_t height;     // 0x36
+                uint16_t min_width;  // 0x38
+                uint16_t max_width;  // 0x3a
+                uint16_t min_height; // 0x3c
+                uint16_t max_height; // 0x3e
                 uint16_t var_40;
                 uint32_t var_42;
                 uint8_t pad_46[0x83E - 0x46];
                 uint16_t var_83E;
-                uint8_t pad_840[0x85A - 0x840];
+                uint8_t pad_840[0x846 - 0x840];
+                uint16_t var_846;
+                uint8_t pad_848[0x854 - 0x848];
+                uint16_t var_854;
+                uint16_t var_856;
+                uint8_t pad_858[0x85A - 0x858];
                 uint16_t var_85A;
                 uint8_t pad_85C[0x870 - 0x85C];
                 uint16_t var_870;
                 uint8_t pad_872[0x882 - 0x872];
```

Comment:
876 word is the `thing` that the viewport is looking at. 0xFFFF for not looking at a `thing`.